Add server-side command support

Ryozuki opened the pull request:
Fixes #2117 

~TODO: Ability to remove commands from the server.~ (done)

Dune-jr commented on src/game/client/components/chat.cpp:
```diff
@@ -24,23 +24,19 @@
+	m_Commands.AddCommand("m", "p", "Mute a player", &Com_Mute);
+	m_Commands.AddCommand("mute", "p", "Mute a player", &Com_Mute);
+	m_Commands.AddCommand("r", "", "Reply to a whisper", &Com_Reply);
+	m_Commands.AddCommand("team", "p", "Switch to team chat", &Com_Team);
```
No parameter for `/team` either

Ryozuki commented on src/game/server/gamecontroller.h:
```diff
@@ -119,6 +120,32 @@ class IGameController
+
+	class CChatCommands
+	{
+		array<CChatCommand*> m_aCommands;
```
the check is already there

Dune-jr replied:
Not that it matters too much, but I think it should be a `if(m_aCommands.size() > MAX_COMMANDS)`

oy replied:
Yeah, would be wetter to use some static array for it. Also on client side. Would provide better memory management for the comment @LordSk mentioned.
Another thing: we use `0` instead of `NULL`.

Dune-jr replied:
@oy my bad, I advised @Ryozuki to replace his `nullptr` with `NULL`s. We have quite a few `NULL` in the code nowadays, isn't there a way to show that it's a pointer type? `0x0`?

oy replied:
@Dune-jr no just use 0 for it. The pointer type you show via the naming sheme `p`.
`NULL` is some old c style and when something crazy happens and it isn't 0 you have some nice debugging times ;)

Ryozuki replied:
Changed NULL to 0.

LordSk commented on src/game/client/components/chat.cpp:
```diff
@@ -1468,50 +1509,104 @@ void CChat::Com_Befriend(CChat *pChatData, const char* pCommand)
+	pCommand->m_pArgsFormat = (char *)mem_alloc(sizeof(char) * (str_length(pArgsFormat) + 1), 1);
+	str_copy(pCommand->m_pArgsFormat, pArgsFormat, str_length(pArgsFormat) + 1);
+
+	pCommand->m_pHelpText = (char *)mem_alloc(sizeof(char) * (str_length(pName) + 1), 1);
```
This should be str_length(pHelpText) (and so your write past the buffer and it crashes). Also I don't see the point of heap allocating everything.

Ryozuki commented:
I changed the arrays to be static, added the ability to remove commands, even vanilla ones.

Fixed a bug where you would keep chat commands after moving to another server.
